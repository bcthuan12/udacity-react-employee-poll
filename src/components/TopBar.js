import { Container, Nav, Navbar } from "react-bootstrap";
import { Link } from "react-router-dom";

const TopBar = () => {
  return (
    <Navbar expand="lg" className="bg-body-tertiary text-start">
      <Container>
        <Nav className="me-auto" variant={"underline"}>
          <Nav.Link as={Link} to="/">
            Home
          </Nav.Link>
          <Nav.Link as={Link} to="/leaderboard">
            Leaderboard
          </Nav.Link>
          <Nav.Link as={Link} to="/new">
            New
          </Nav.Link>
        </Nav>
      </Container>
    </Navbar>
  );
};
export default TopBar;
