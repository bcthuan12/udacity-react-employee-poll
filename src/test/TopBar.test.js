import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import TopBar from "../components/TopBar";
import { store } from "../index";
import { BrowserRouter } from "react-router-dom";

describe("Topbar Test suite", () => {
  test("Should match snapshot", () => {
    const comp = render(
      <Provider store={store}>
        <BrowserRouter>
          <TopBar />
        </BrowserRouter>{" "}
      </Provider>,
    );
    expect(comp).toMatchSnapshot();
  });

  test("Should render home menu", () => {
    const comp = render(
      <Provider store={store}>
        <BrowserRouter>
          <TopBar />
        </BrowserRouter>
      </Provider>,
    );
    expect(screen.getByTestId("homeMenu")).toBeInTheDocument();
  });

  test("Should render leardboard menu", () => {
    const comp = render(
      <Provider store={store}>
        <BrowserRouter>
          <TopBar />
        </BrowserRouter>
      </Provider>,
    );
    expect(screen.getByTestId("leaderboardMenu")).toBeInTheDocument();
  });

  test("Should render new menu", () => {
    const comp = render(
      <Provider store={store}>
        <BrowserRouter>
          <TopBar />
        </BrowserRouter>
      </Provider>,
    );
    expect(screen.getByTestId("newMenu")).toBeInTheDocument();
  });
});
