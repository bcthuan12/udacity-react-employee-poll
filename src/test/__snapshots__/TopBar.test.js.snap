// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar Test suite Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="bg-body-tertiary justify-content-between navbar navbar-expand-lg navbar-light"
      >
        <div
          class="m-lg-2 row"
        >
          <div
            class="me-auto navbar-nav nav-underline"
          >
            <a
              class="nav-link"
              data-testid="homeMenu"
              href="/"
            >
              Home
            </a>
            <a
              class="nav-link"
              data-testid="leaderboardMenu"
              href="/leaderboard"
            >
              Leaderboard
            </a>
            <a
              class="nav-link"
              data-testid="newMenu"
              href="/add"
            >
              New
            </a>
          </div>
        </div>
        <div>
          <img
            class=""
            style="width: 60px;"
          />
          <span
            class="mb-1 text-muted m-lg-2"
            style="font-size: 20px;"
          />
          <button
            class="p-1 m-lg-2 btn btn-warning"
            type="button"
          >
            Logout
          </button>
        </div>
      </nav>
       
    </div>
  </body>,
  "container": <div>
    <nav
      class="bg-body-tertiary justify-content-between navbar navbar-expand-lg navbar-light"
    >
      <div
        class="m-lg-2 row"
      >
        <div
          class="me-auto navbar-nav nav-underline"
        >
          <a
            class="nav-link"
            data-testid="homeMenu"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-link"
            data-testid="leaderboardMenu"
            href="/leaderboard"
          >
            Leaderboard
          </a>
          <a
            class="nav-link"
            data-testid="newMenu"
            href="/add"
          >
            New
          </a>
        </div>
      </div>
      <div>
        <img
          class=""
          style="width: 60px;"
        />
        <span
          class="mb-1 text-muted m-lg-2"
          style="font-size: 20px;"
        />
        <button
          class="p-1 m-lg-2 btn btn-warning"
          type="button"
        >
          Logout
        </button>
      </div>
    </nav>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
